src/Build/Evaluation/LazyItemEvaluator.RemoveOperation.cs

```diff
@@ -34,6 +39,20 @@ protected override ImmutableList<I> SelectItems(ImmutableList<ItemData>.Builder
                     new BuildEventFileInfo(string.Empty),
                     "OM_MatchOnMetadataIsRestrictedToOnlyOneReferencedItem");
 
+                if (_matchOnMetadata.IsEmpty && ItemspecContainsASingleItemReference(_itemSpec, _itemElement.ItemType))
+                {
+                    // Perf optimization: If the Remove operation references itself (e.g. <I Remove="@(I)"/>)
+                    // then all items are removed and matching is not necessary
+                    listBuilder.RemoveAll((_) => true);
```

Comment:
nit:
```suggestion
                    listBuilder.Clear();
```

Reply:
🤦‍♂️ I swear I used and/or looked for this before. Thanks!